File: backend/internal/models/reservation.go
package models

import "time"

type ReservationStatus string

const (
	StatusPending   ReservationStatus = "pending"
	StatusConfirmed ReservationStatus = "confirmed"
	StatusCancelled ReservationStatus = "cancelled"
	StatusCompleted ReservationStatus = "completed"
)

type PaymentStatus string

const (
	PaymentNone      PaymentStatus = "none"
	PaymentPending   PaymentStatus = "pending"
	PaymentSucceeded PaymentStatus = "succeeded"
	PaymentFailed    PaymentStatus = "failed"
	PaymentRefunded  PaymentStatus = "refunded"
)

type Reservation struct {
	Base
	UserID          uint              `gorm:"not null" json:"userId"`
	User            User              `json:"user,omitempty"`
	EstablishmentID uint              `gorm:"not null" json:"establishmentId"`
	Establishment   Establishment     `json:"establishment,omitempty"`
	CheckInDate     time.Time         `gorm:"not null" json:"checkInDate"`
	CheckOutDate    *time.Time        `json:"checkOutDate"`
	GuestsCount     int               `gorm:"default:1" json:"guestsCount"`
	TotalPriceFCFA  int64             `json:"totalPriceFcfa"`
	Status          ReservationStatus `gorm:"default:pending" json:"status"`
	SpecialRequests string            `json:"specialRequests"`
	PaymentIntentID string            `gorm:"index" json:"paymentIntentId,omitempty"`
	PaymentStatus   PaymentStatus     `gorm:"default:none" json:"paymentStatus"`
}

File: mobile/lib/models/reservation.dart
class Reservation {
  final int id;
  final int establishmentId;
  final String establishmentName;
  final String checkInDate;
  final String checkOutDate;
  final int guestsCount;
  final String specialRequests;
  final String status;
  final String createdAt;

  const Reservation({
    required this.id,
    this.establishmentId = 0,
    this.establishmentName = '',
    this.checkInDate = '',
    this.checkOutDate = '',
    this.guestsCount = 1,
    this.specialRequests = '',
    this.status = 'pending',
    this.createdAt = '',
  });

  factory Reservation.fromJson(Map<String, dynamic> json) => Reservation(
        id: json['id'] ?? json['ID'] ?? 0,
        establishmentId: json['establishmentId'] ?? 0,
        establishmentName:
            json['establishment']?['name'] ?? json['establishmentName'] ?? '',
        checkInDate: json['checkInDate'] ?? '',
        checkOutDate: json['checkOutDate'] ?? '',
        guestsCount: json['guestsCount'] ?? 1,
        specialRequests: json['specialRequests'] ?? '',
        status: json['status'] ?? 'pending',
        createdAt: json['createdAt'] ?? json['CreatedAt'] ?? '',
      );
}
